### PR TITLE
Feature/ctv 2621 html5 a b testing support

### PR DESCRIPTION
**Pull request by rblaa**
### JIRA
https://truextech.atlassian.net/browse/CTV-2621

### Description
Enable A/B Testing Support, both via multiple choice cards (for variant event tracking) and testVariations in bluescript (via the ad[0].variations values).

### Testing
* Run from Skyline
* Run unit tests.

### Commit Message Subject(s)
CTV-2621: HTML5 - A/B Testing Support

### Additional Context
n/a

### Related PRs
truex-shared: https://github.com/socialvibe/truex-shared-js/pull/28/files
truex-bluescript: https://github.com/socialvibe/truex-bluescript-js/pull/4
C3: https://github.com/socialvibe/container_core/pull/363
IC: https://github.com/socialvibe/integration_core/pull/131
TAR HTML5: https://github.com/socialvibe/TruexAdRenderer-HTML5/pull/65
Skyline: https://github.com/socialvibe/Skyline/pull/111
firetv-webview: https://github.com/socialvibe/firetv-webview-webapp/pull/30

### Checks
- [x] Includes unit test coverage _(if applicable)_
- [x] Includes functional test coverage _(at feature-level, if applicable)_
- [x] Proposed commit messages follow [team conventions](https://github.com/socialvibe/adlabs-wiki/blob/develop/git/README.md#commits)

### CODE REVIEW

File: src/utils/ab-testing.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *   VAST config's user.id property if missing.
+ * @return {Object} The choice card configuration chosen as the test variant, or null if multiple test
+ *   variants are not present.
+ */
+export function getChoiceCardTestConfiguration(vastConfig, userId) {
+    if (hasChoiceCardTestVariants(vastConfig)) {
+        if (!userId) userId = vastConfig.user && vastConfig.user.id;
+        if (userId) {
+            // Hash the user id via the sum of its char codes.
+            // Prevent sum from getting too large, using (A + B) mod N = (A mod N + B mod N) mod N
```
Comment (kylelamx) [inline src/utils/ab-testing.js]: why even do this? 
`userId.charCodeAt(i)` is no where near "too large"

**Review by kylelamx — COMMENTED** [review]
I'm not too sure I understand why these choice card codes are in the shared repo.

File: src/utils/ab-testing.js (hunk 1 above)
Comment (rblaa) [reply src/utils/ab-testing.js]: not the individual char code of course, but the fact that the loop is unbounded.
given the mod equivalence, this approach is theoretically better, no downsides.

**Comment by rblaa** [comment]
> I'm not too sure I understand why these choice card codes are in the shared repo.

Because the same algorithm is used both in the test card config and integration_core to decide what the test variant choice card config is to be.

UPDATE: I think i can generalize things, renaming the method to just getHashedIndex(id, n), so that it is more generic for a shared repo. The caller impact is minimal

File: src/utils/ab-testing.js (hunk 1 above)
Comment (rblaa) [reply src/utils/ab-testing.js]: actually you aright. there is no way this will overflow in practice, so simpler code is better

**Review by kylelamx — APPROVED** [review]
love it

**Pull request merged**